File: java/src/main/java/io/hron/ast/MonthRepeat.java
package io.hron.ast;

import java.util.List;

/**
 * A month-based repeat expression like "every month on the 1st at 9:00".
 *
 * @param interval the number of months between occurrences (1 for every month)
 * @param target the day(s) within the month to fire
 * @param times the times of day to fire
 */
public record MonthRepeat(int interval, MonthTarget target, List<TimeOfDay> times)
    implements ScheduleExpr {
  /** Creates a new MonthRepeat with defensive copy of times list. */
  public MonthRepeat {
    times = List.copyOf(times);
  }
}

File: csharp/Hron/Ast/MonthRepeat.cs
namespace Hron.Ast;

/// <summary>
/// A month-based repeat expression like "every month on the 1st at 9:00".
/// </summary>
/// <param name="Interval">The number of months between occurrences (1 for every month)</param>
/// <param name="Target">The day(s) within the month to fire</param>
/// <param name="Times">The times of day to fire</param>
public sealed record MonthRepeat(int Interval, MonthTarget Target, IReadOnlyList<TimeOfDay> Times) : IScheduleExpr;
